lib/hexo/post.js

```diff
@@ -259,6 +259,9 @@ class Post {
       }
     }
 
+    // front-matter overrides renderer's option
+    disableNunjucks = typeof data.disableNunjucks === 'boolean' ? data.disableNunjucks : disableNunjucks;
```

Comment:
Could be done before checking the renderer.

Reply:
front-matter should override renderer in this case.